server/src/test/java/org/elasticsearch/index/mapper/NumberFieldTypeTests.java

```diff
@@ -346,6 +346,55 @@ public void testLongRangeQueryWithDecimalParts() {
         );
     }
 
+    public void testHalfFloatRangeQueryWithOverflowingBounds() {
+        MappedFieldType ft = new NumberFieldMapper.NumberFieldType("field", NumberType.HALF_FLOAT, randomBoolean());
+        final float min_half_float = -65504;
+        final float max_half_float = 65504;
+        assertEquals(
+            ft.rangeQuery(min_half_float, 10, true, true, null, null, null, MOCK_CONTEXT),
+            ft.rangeQuery(-1e+300, 10, true, true, null, null, null, MOCK_CONTEXT)
+        );
```

Comment:
Please add tests where `includeLower` and `includeUpper` are false.

Reply:
done:

https://github.com/elastic/elasticsearch/blob/988d2915493d113039ff2cd91888f7a28814ce29/server/src/test/java/org/elasticsearch/index/mapper/NumberFieldTypeTests.java#L373-381

https://github.com/elastic/elasticsearch/blob/988d2915493d113039ff2cd91888f7a28814ce29/server/src/test/java/org/elasticsearch/index/mapper/NumberFieldTypeTests.java#L407-L415

plus more in the yaml tests